### ISSUE TITLE
2010-04-23: A Daemon for LaTeXML

### ISSUE THREAD

**Issue by dginev** [brucemiller/LaTeXML#128]
[Originally Ticket 1452]

It is my pleasure to announce a first implementation of a LaTeXML daemon, realized on the [arXMLiv branch](https://trac.mathweb.org/LaTeXML/browser/branches/arXMLiv) of the [LaTeXML repository](https://trac.mathweb.org/LaTeXML/browser/).

The daemon has the purpose of providing an efficient and convenient way for performing vast conversion batch jobs of TeX sources. In contrast to LaTeXML, which needs to be invoked individually on every source, the daemon is initialized once and can process an arbitrary amount of sources on demand.

= Features: =
- Can emulate the behavior of latexml, latexmlmath and latexmlpost
- Robust against malformed TeX - persists against normal and fatal errors caused by the TeX source.
- Has three levels of speedup, depending on the converted inputs:
  1. **math** mode : Fastest, one formula per input line
  2. **fragment** mode : Faster, expects document fragments based on a homogeneous document class and/or packages (as filenames).
  3. **standard**  mode : Slowest of the three, can process any heterogeneous collection of TeX documents (as filenames). It is still faster than the original "latexml" binary, as perl is initialized only once for the entire batch.
- Easily customizable log and destination targets via the #dir,#name and #ext filepath patterns.
- An "autoflush" switch to automatically restart the daemon after a certain amount of converted input (keeps memory consumption low)
- A new "keepTeX" switch which allows to preserve the TeX formula source as a MathML annotation.

= Download and Install: =
You should follow the standard instructions for [installing LaTeXML from a repository](http://dlmf.nist.gov/LaTeXML/get.html#SS4).

However, instead of checking out the trunk, you should check out the arXMLiv branch:

```
svn co https://svn.mathweb.org/repos/LaTeXML/branches/arXMLiv
```

Then you can continue with the standard installation from source. The daemon binary is located at `bin/latexmld`

= ToDo and Caveats: = 
- The presented above software is an early first release, so use the daemon at your own risk. It has already been put to production use by the arXMLiv project and has shown stability in converting 100,000 LaTeX sources so far. However, no liability is taken for the provided software, proceed to use with caution.
- Documentation is lacking examples and detailed use-cases for the moment, soon to come!
- The --keepXMath and --openmath switches are yet to be supported, also some non-standard combinations of the switches may not work as expected. Eventually, all latexml and latexmlpost capabilities will be available within the daemon.
- Since the eventual goal of the daemon is to enter the standard distribution of LaTeXML, it is released in the Public Domain.

= How you can help: =
Install the daemon, try it out and give us feedback! For any bug reports, feature requests and general usability comments, please open a new Ticket under the "latexmld" component at the LaTeXML Trac (https://trac.mathweb.org/LaTeXML/). You can also leave comments on this news item itself.

We are looking forward to your first impressions!
